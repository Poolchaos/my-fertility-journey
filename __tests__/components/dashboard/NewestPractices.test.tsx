describe('NewestPractices', () => {
  test.todo('Re-enable tests for NewestPractices');
});
// import React from 'react';
// import {
//   render,
//   screen,
//   fireEvent,
//   within,
// } from '@testing-library/react-native';
// import NewestPractices from '../../../components/features/dashboard/NewestPractices';
// import { newestPracticesData } from '../../../components/features/dashboard/newestPractices.data';

// const mockToggleStatus = jest.fn();
// const mockEdit = jest.fn();
// const mockDelete = jest.fn();

// describe('NewestPractices Component', () => {
//   beforeEach(() => {
//     jest.clearAllMocks();
//   });

//   it('renders correctly with data', () => {
//     render(
//       <NewestPractices
//         data={newestPracticesData}
//         onToggleStatus={mockToggleStatus}
//         onEdit={mockEdit}
//         onDelete={mockDelete}
//       />
//     );

//     expect(screen.getByText('Newest Practices')).toBeTruthy();

//     newestPracticesData.forEach((practice) => {
//       const practiceElements = screen.getAllByText(practice.name);
//       expect(practiceElements.length).toBeGreaterThan(0);

//       practiceElements.forEach((practiceElement) => {
//         const practiceContainer = practiceElement.parent;
//         expect(practiceContainer).not.toBeNull();

//         expect(
//           within(practiceContainer!).getByText(practice.phone)
//         ).toBeTruthy();
//         expect(
//           within(practiceContainer!).getByText(practice.email)
//         ).toBeTruthy();
//       });
//     });
//   });

//   it('calls toggle status callback', () => {
//     render(
//       <NewestPractices
//         data={newestPracticesData}
//         onToggleStatus={mockToggleStatus}
//         onEdit={mockEdit}
//         onDelete={mockDelete}
//       />
//     );

//     const switches = screen.getAllByRole('switch');
//     expect(switches.length).toBeGreaterThan(0);

//     fireEvent.press(switches[0]);
//     expect(mockToggleStatus).toHaveBeenCalledTimes(1);
//   });

//   it('calls edit callback', () => {
//     render(
//       <NewestPractices
//         data={newestPracticesData}
//         onToggleStatus={mockToggleStatus}
//         onEdit={mockEdit}
//         onDelete={mockDelete}
//       />
//     );

//     const editButtons = screen.getAllByTestId(/^edit-button-/);
//     expect(editButtons.length).toBeGreaterThan(0);

//     fireEvent.press(editButtons[0]);
//     expect(mockEdit).toHaveBeenCalledTimes(1);
//   });

//   it('calls delete callback', () => {
//     render(
//       <NewestPractices
//         data={newestPracticesData}
//         onToggleStatus={mockToggleStatus}
//         onEdit={mockEdit}
//         onDelete={mockDelete}
//       />
//     );

//     const deleteButtons = screen.getAllByTestId(/^delete-button-/);
//     expect(deleteButtons.length).toBeGreaterThan(0);

//     fireEvent.press(deleteButtons[0]);
//     expect(mockDelete).toHaveBeenCalledTimes(1);
//   });

//   it('matches snapshot', () => {
//     const { toJSON } = render(
//       <NewestPractices
//         data={newestPracticesData}
//         onToggleStatus={mockToggleStatus}
//         onEdit={mockEdit}
//         onDelete={mockDelete}
//       />
//     );
//     expect(toJSON()).toMatchSnapshot();
//   });
// });
