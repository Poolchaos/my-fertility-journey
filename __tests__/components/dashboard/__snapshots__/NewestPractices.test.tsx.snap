// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewestPractices Component matches snapshot 1`] = `
<View
  style={
    [
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
      {
        "backgroundColor": "#fff",
        "boxShadow": "0px 0px 20px rgba(0, 0, 0, 0.1)",
      },
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "700",
        "lineHeight": 28,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    Newest Practices
  </Text>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginBottom": 16,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "700",
            "lineHeight": 28,
            "marginBottom": 8,
          }
        }
      >
        Cape Fertility Clinic
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        +21 794 3956
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        info@capefertility.co.za
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
            "marginBottom": 8,
          }
        }
      >
        Created: 
        04/10/2021
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#CEE4E8"
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            thumbTintColor="#67adb9"
            tintColor="#ccc"
            value={true}
          />
          <Text
            style={
              {
                "marginLeft": 8,
              }
            }
          >
            Active
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginRight": 12,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/images/pencil.svg",
                }
              }
              style={
                {
                  "height": 14,
                  "width": 14,
                }
              }
              tintColor="#67adb9"
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/images/trash.svg",
                }
              }
              style={
                {
                  "height": 14,
                  "width": 14,
                }
              }
              tintColor="#67adb9"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginBottom": 16,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "700",
            "lineHeight": 28,
            "marginBottom": 8,
          }
        }
      >
        Cape Fertility Clinic
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        +21 794 3956
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        info@capefertility.co.za
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
            "marginBottom": 8,
          }
        }
      >
        Created: 
        04/10/2021
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#CEE4E8"
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            thumbTintColor="#67adb9"
            tintColor="#ccc"
            value={true}
          />
          <Text
            style={
              {
                "marginLeft": 8,
              }
            }
          >
            Active
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginRight": 12,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/images/pencil.svg",
                }
              }
              style={
                {
                  "height": 14,
                  "width": 14,
                }
              }
              tintColor="#67adb9"
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/images/trash.svg",
                }
              }
              style={
                {
                  "height": 14,
                  "width": 14,
                }
              }
              tintColor="#67adb9"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginBottom": 16,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "700",
            "lineHeight": 28,
            "marginBottom": 8,
          }
        }
      >
        Cape Fertility Clinic
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        +21 794 3956
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
      >
        info@capefertility.co.za
      </Text>
      <Text
        style={
          {
            "color": "rgb(75 85 99 / var(--tw-text-opacity, 1))",
            "fontSize": 14,
            "lineHeight": 20,
            "marginBottom": 8,
          }
        }
      >
        Created: 
        04/10/2021
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#CEE4E8"
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            thumbTintColor="#fff"
            tintColor="#ccc"
            value={false}
          />
          <Text
            style={
              {
                "marginLeft": 8,
              }
            }
          >
            Disabled
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginRight": 12,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/images/pencil.svg",
                }
              }
              style={
                {
                  "height": 14,
                  "width": 14,
                }
              }
              tintColor="#67adb9"
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/images/trash.svg",
                }
              }
              style={
                {
                  "height": 14,
                  "width": 14,
                }
              }
              tintColor="#67adb9"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 16,
        "marginRight": 16,
        "marginTop": 16,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "rgb(59 130 246 / var(--tw-text-opacity, 1))",
          },
          {
            "color": "#67adb9",
          },
        ]
      }
    >
      See All
    </Text>
    <Image
      source={
        {
          "testUri": "../../../assets/images/caret-right.svg",
        }
      }
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
      tintColor="#67adb9"
    />
  </View>
</View>
`;
